project/ScapegoatInspectionsGenerator.scala

```diff
@@ -69,11 +69,20 @@ object ScapegoatInspectionsGenerator {
 
     val lines = List(
       "package com.mwz.sonar.scala.scapegoat.inspections",
-      "sealed trait Level",
+      "import org.sonar.api.batch.rule.Severity",
+      "sealed trait Level {",
+      "/** Returns the SonarQube rule severity associated with this Level */",
+      "def toRuleSeverity: Severity",
+      "}",
       "object Level {",
-      "case object Error extends Level",
-      "case object Warning extends Level",
-      "case object Info extends Level",
+      "case object Error extends Level { override def toRuleSeverity: Severity = Severity.MAJOR }",
+      "case object Warning extends Level { override def toRuleSeverity: Severity = Severity.MINOR }",
+      "case object Info extends Level { override def toRuleSeverity: Severity = Severity.INFO }",
+      "def fromName(name: String): Level = name.toLowerCase match {",
+      "case \"error\" => Error",
+      "case \"warning\" => Warning",
+      "case \"info\" => Info",
+      "}",
```

Comment:
I realise that having to maintain code which is in a list format isn't the best, maybe we could extract it into a file which you can then load before processing?

Reply:
I totally agree, but I think is better to make that change in another PR after this one gets merged. What do you think?

Reply:
yeah, I'm totally fine with that.